src/components/AddComment.jsx

```diff
@@ -1,9 +1,61 @@
-const AddComment = () => {
+import { useState } from "react";
+import { postComment } from "../api";
+
+const AddComment = ({article_id}) => {
+
+    const [commentInput, setCommentInput] = useState("")
+    const [addedCommentList, setAddedCommentList] = useState([])
+    const [dataValid, setDataValid] = useState(true)
+    const [isPosting,setIsPosting] = useState(false)
+    const [isPosted,setIsPosted] = useState(false)
+
+
+    const commentPost = (e) => {
+        e.preventDefault();
+        if (commentInput.length === 0) {
+            e.preventDefault();
+            setDataValid(false)
+            setIsPosted(false)
+        } else {
+            setDataValid(true)
+            setIsPosting(true)
+            postComment(article_id,commentInput)
+            .then(res => {
+                setIsPosting(false)
+                setIsPosted(true)
+                setAddedCommentList((currList) => [...currList,res.comment])
+            })
+            setCommentInput("")
+        }
+    }
+
     return (
-        <form className="AddComment">
-            <input type="text" placeholder="Comment..." className="AddCommentInput"></input>
+        <div>
+            <form className="AddComment" onSubmit={commentPost}>
+            <input type="text" placeholder="Comment..." value={commentInput} className="AddCommentInput" onChange={(e) => setCommentInput(e.target.value)}></input>
+            {dataValid?null:<p>Please enter a comment</p>}
+            {isPosting?<p>Posting....</p>:null}
```

Comment:
Two bits of feedback here: I would recommend using the textarea html tag instead of an input tag. This is because an input tag doesn't let us have multiple lines, where as the textarea does.

Also you can add a word inside the tag called 'required' which will not let the user progress without putting something in the box. That way you won't need state like dataValid and checking if comments length is greater than 0 and so on.

Here's the w3schools page for text area: [https://www.w3schools.com/tags/tag_textarea.asp](url) You can see it accepts an attribute called required

w3 schools page for the required attribute: [https://www.w3schools.com/tags/att_textarea_required.asp](url)